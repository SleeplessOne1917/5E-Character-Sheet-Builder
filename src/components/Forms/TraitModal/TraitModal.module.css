.modal {
    width: 100%;
    height: 100%;
    background-color: var(--heaven);
    display: grid;
    grid-template-columns: 100%;
    grid-template-rows: 1fr 4fr 1fr;
    grid-template-areas: "title"
                        "content"
                        "buttons";
    font-family: var(--font-display);
    overflow-x: hidden;
}

.title {
    grid-area: title;
    font-size: 2rem;
    text-align: center;
    padding: 1rem;
}

.content {
    grid-area: content;
    padding: 2.5rem;
}

.buttons {
    grid-area: buttons;
    display: flex;
}

@media screen and (min-width: 768px) {
    .modal {
        height: 80%;
        width: 80%;
    }
}

@media screen and (min-width: 1024px) {
    .modal {
        height: 60%;
        width: 60%;
    }
}