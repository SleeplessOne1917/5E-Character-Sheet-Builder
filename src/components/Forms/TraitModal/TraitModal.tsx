'use client';

import Button from '../../Button/Button';
import { CSSProperties } from 'react';
import LoadingSpinner from '../../LoadingSpinner/LoadingSpinner';
import ModalBackground from '../../ModalBackground/ModalBackground';
import Trait from '../../../types/trait';
import classes from './TraitModal.module.css';
import dynamic from 'next/dynamic';

const MarkdownParser = dynamic(
	() => import('../../MarkdownParser/MarkdownParser'),
	{
		loading: () => (
			<div
				style={{
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'center'
				}}
			>
				<LoadingSpinner />
			</div>
		)
	}
);

const buttonStyle: CSSProperties = {
	flexGrow: 1,
	display: 'flex',
	justifyContent: 'center',
	alignItems: 'center',
	fontSize: '2rem',
	border: 0,
	margin: 0,
	borderRadius: 0,
	width: '50%'
};

type TraitModalProps = {
	show: boolean;
	onClose: () => void;
	onAction: () => void;
	mode: 'omit' | 'include';
	trait: Pick<Trait, 'name' | 'description'>;
};

const TraitModal = ({
	show,
	onAction,
	onClose,
	mode,
	trait
}: TraitModalProps) => {
	return (
		<ModalBackground show={show} testId="trait-modal">
			<div className={classes.modal}>
				<div className={classes.title}>{trait.name}</div>
				<div className={classes.content}>
					{show && <MarkdownParser input={trait.description} />}
				</div>
				<div className={classes.buttons}>
					<Button onClick={onAction} style={buttonStyle} positive>
						{mode === 'include' ? 'Include' : 'Omit'} Trait
					</Button>
					<Button onClick={onClose} style={buttonStyle}>
						Close
					</Button>
				</div>
			</div>
		</ModalBackground>
	);
};

export default TraitModal;
