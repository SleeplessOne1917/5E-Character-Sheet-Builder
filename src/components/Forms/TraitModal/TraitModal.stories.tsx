import { ComponentMeta, ComponentStory } from '@storybook/react';

import TraitModal from './TraitModal';

export default {
	title: 'Components/Forms/TraitModal',
	component: TraitModal,
	args: {
		show: true,
		trait: {
			name: 'Foo the Bar',
			description:
				'Your natrual fooness gives you advantage on saving throws against bar.'
		}
	}
} as ComponentMeta<typeof TraitModal>;

const Template: ComponentStory<typeof TraitModal> = args => (
	<TraitModal {...args} />
);

export const Include = Template.bind({});
Include.args = {
	mode: 'include'
};

export const Omit = Template.bind({});
Omit.args = {
	mode: 'omit'
};
