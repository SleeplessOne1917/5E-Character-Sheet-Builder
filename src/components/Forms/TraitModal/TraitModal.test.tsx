import * as stories from './TraitModal.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/testing-react';

const { Include, Omit } = composeStories(stories);

describe('redners correctly', () => {
	it('include', () => {
		render(<Include />);

		expect(screen.getByTestId('trait-modal')).toMatchSnapshot();
	});

	it('omit', () => {
		render(<Omit />);

		expect(screen.getByTestId('trait-modal')).toMatchSnapshot();
	});
});
