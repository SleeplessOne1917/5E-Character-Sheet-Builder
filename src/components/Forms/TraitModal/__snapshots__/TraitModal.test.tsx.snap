// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redners correctly include 1`] = `
<div
  class="modal-container"
  data-testid="trait-modal"
  style="display: flex;"
>
  <div
    class="modal"
  >
    <div
      class="title"
    >
      Foo the Bar
    </div>
    <div
      class="content"
    >
      <div
        style="display: flex; align-items: center; justify-content: center;"
      >
        <div
          class="container"
        >
          <svg
            class="icon"
          >
            <use
              xlink:href="/Icons.svg#spinner"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button positive"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0; width: 50%;"
        type="button"
      >
        Include
         Trait
      </button>
      <button
        class="button negative"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0; width: 50%;"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`redners correctly omit 1`] = `
<div
  class="modal-container"
  data-testid="trait-modal"
  style="display: flex;"
>
  <div
    class="modal"
  >
    <div
      class="title"
    >
      Foo the Bar
    </div>
    <div
      class="content"
    >
      <div
        class="markdown"
      >
        <p
          class="paragraph"
        >
          Your natrual fooness gives you advantage on saving throws against bar.
        </p>
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button positive"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0; width: 50%;"
        type="button"
      >
        Omit
         Trait
      </button>
      <button
        class="button negative"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0; width: 50%;"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
